include/cliffordsynthesis/CliffordSynthesizer.hpp

```diff
@@ -0,0 +1,289 @@
+/*
+* This file is part of the MQT QMAP library which is released under the MIT license.
+* See file README.md or go to https://iic.jku.at/eda/research/ibm_qx_mapping/ for more information.
+*/
+
+#ifndef QMAP_CLIFFORDSYNTHESIS_H
+#define QMAP_CLIFFORDSYNTHESIS_H
+
+#include "Architecture.hpp"
+#include "CliffordOptimizationResult.hpp"
+#include "Encodings/Encodings.hpp"
+#include "LogicBlock/LogicBlock.hpp"
+#include "QuantumComputation.hpp"
+#include "operations/OpType.hpp"
+#include "operations/StandardOperation.hpp"
+
+#include <algorithm>
+#include <bitset>
+#include <cassert>
+#include <chrono>
+#include <cmath>
+#include <functional>
+#include <iostream>
+#include <istream>
+#include <iterator>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <sstream>
+#include <thread>
+#include <tuple>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+class CliffordOptimizer {
+public:
+    CliffordOptimizer() = default;
+    void init(bool pchoose_best = false, bool puse_embedding = false, unsigned char pnqubits = 0, unsigned short pinitial_timesteps = 0, SynthesisStrategy strategy = SynthesisStrategy::UseMinimizer, SynthesisTarget ptarget = SynthesisTarget::GATES);
+    void setCircuit(const qc::QuantumComputation& qc);
+    void setTableau(Tableau& targetTabl);
+    void optimize();
+
+    void setArchitecture(const Architecture& arch) {
+        architecture = arch;
+        if (nqubits == 0) {
+            nqubits = architecture.getNqubits();
+        }
+        auto map = highestFidelityMap.emplace_back();
+        architecture.getHighestFidelityCouplingMap(nqubits, map);
+    }
+    bool                   choose_best   = false;
+    bool                   use_embedding = false;
+    unsigned char          nqubits       = 0U;
+    std::set<signed char>  used_qubits{};
+    unsigned short         initial_timesteps = 0U;
+    int                    verbose           = 0;
+    int                    nthreads          = 1;
+    SynthesisStrategy      strategy          = SynthesisStrategy::UseMinimizer;
+    SynthesisTarget        target            = SynthesisTarget::GATES;
+    SynthesisMethod        method            = SynthesisMethod::Z3;
+    qc::QuantumComputation circuit;
+
+    std::vector<CouplingMap> highestFidelityMap;
+
+    Tableau initialTableau{};
+    Tableau targetTableau{};
+    Tableau modelTableau{};
+
+    void dumpResult(const std::string& outputFilename) {
+        if (optimal_results.resultCircuit.empty()) {
+            std::cerr << "Circuit is empty." << std::endl;
+            return;
+        }
+
+        size_t      dot       = outputFilename.find_last_of('.');
+        std::string extension = outputFilename.substr(dot + 1U);
+        std::transform(extension.begin(), extension.end(), extension.begin(), [](unsigned char c) { return ::tolower(c); });
+        if (extension == "real") {
+            dumpResult(outputFilename, qc::Real);
```

Comment:
If I recall correctly, only Qasim export is actually implemented. Accordingly, this can be removed